ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -1130,6 +1133,26 @@ export const WorkspaceEdit = fp.flow(
       this.saveWorkspace();
     }
 
+    getAskAboutAIAN() {
+      const publicCDRVersionNumber: number = getCdrVersion(
+        this.state.workspace,
+        this.props.cdrVersionTiersResponse
+      ).publicReleaseNumber;
+
+      return publicCDRVersionNumber >= 8;
+    }
+
+    /* Deeply clones the provided workspace and makes any adjustments necessary without
+     * adjusting local state. */
+    preProcessWorkspace(workspace: Workspace) {
+      const adjustedWorkspace = JSON.parse(JSON.stringify(workspace));
```

Comment:
is the JSON roundtrip necessary?  Could we do something like
```
{
  ...workspace,
  researchPurpose: {
    ...workspace.researchPurpose,
    newValues
  }
}
```
instead?

Reply:
looks like a good solution to the problem though

Reply:
Thanks! The double spread hadn't come to mind. I know that a single spread would shallow copy nested attributes, like those in researchPurpose, but your suggestion above is great. Thank you!